File: frontend/src/api/survey-observation.ts
import { api } from './client'
import type { ApiEnvelope } from '@/types/api'
import type { SurveyObservation } from '@/types/survey-observation'

export interface ObservationQuery {
  parcel_id?: number
  state?: string
  page?: number
  page_size?: number
}

export interface ObservationImportInput {
  parcel_id: number
  observation_code: string
  point_geojson: string
  observed_at: string
  method: string
  horizontal_accuracy_m: number
  source_checksum: string
  observation_state?: string
  quality_note?: string
}

export type ObservationState = 'accepted' | 'rejected' | 'superseded'

export interface ObservationTransitionInput {
  to: ObservationState
  version: number
  replacement_observation_id?: number
  quality_note?: string
}

export const surveyObservationApi = {
  list: (params?: ObservationQuery) => api.get<ApiEnvelope<SurveyObservation[]>>('/observations', { params }),
  detail: (id: number) => api.get<ApiEnvelope<SurveyObservation>>(`/observations/${id}`),
  import: (body: ObservationImportInput) => api.post<ApiEnvelope<SurveyObservation>>('/observations/import', body),
  transition: (id: number, body: ObservationTransitionInput) =>
    api.post<ApiEnvelope<SurveyObservation>>(`/observations/${id}/transition`, body),
}

File: backend/internal/dto/survey_observation.go
package dto

import "time"

type ObservationQuery struct {
	ParcelID *uint
	State    string
	Page     int
	PageSize int
}

type ImportObservationRequest struct {
	ParcelID            uint      `json:"parcel_id" validate:"required,gt=0"`
	ObservationCode     string    `json:"observation_code" validate:"required,min=2,max=80"`
	PointGeoJSON        string    `json:"point_geojson" validate:"required"`
	ObservedAt          time.Time `json:"observed_at" validate:"required"`
	Method              string    `json:"method" validate:"required,min=2,max=48"`
	HorizontalAccuracyM float64   `json:"horizontal_accuracy_m" validate:"required,gt=0,lte=1000"`
	SourceChecksum      string    `json:"source_checksum" validate:"required,min=8,max=128"`
	ObservationState    string    `json:"observation_state" validate:"omitempty,oneof=accepted rejected superseded"`
	QualityNote         string    `json:"quality_note" validate:"max=1000"`
}

type ObservationTransitionRequest struct {
	To                       string `json:"to" validate:"required,oneof=accepted rejected superseded"`
	Version                  uint   `json:"version" validate:"required,gt=0"`
	ReplacementObservationID *uint  `json:"replacement_observation_id" validate:"omitempty,gt=0"`
	QualityNote              string `json:"quality_note" validate:"max=1000"`
}
